### PR TITLE
Feature/lliss/add api

### PR DESCRIPTION
**Pull request by lliss**
Adds a REST based API that can be used to query data from the postgres database.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -59,7 +67,44 @@ class myPlugin {
...
 }
 
-module.exports = myPlugin;
+module.exports = myDbPlugin;
+```
+
+An API plugin will also receive an object that includes a knex connection and a logger. The API plugin must implement an `addRoutes` method that will be handed a restify `server` object and will append new routes to it. A valid api plugin might look like the following:
+
+```javascript
+class myApiPlugin {
```
Comment (tortillaj) [inline README.md]: Yes! Awesome 😀

File: lib/Api.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+      next(e);
+    }
+
+    self.db.getBuildById(buildId)
+      .then(function(result) {
+        res.send(result);
+        next();
+      })
+      .catch(function(error) {
+        self.logger.error(error);
```
Comment (tortillaj) [inline lib/Api.js]: Can we pick one way to name an error? `e` in line 86 or `error` here in 96.

File: lib/Db.js [hunk 3]
```diff
@@ -21,6 +21,35 @@ class Db {
...
+      .where({buildId});
+  }
+
+  getBuildSize(buildId) {
+    return this.getBuildById(buildId)
+      .then(function(result) {
+        return result.bytesReal;
+      });
+  }
+
```
Comment (tortillaj) [inline lib/Db.js]: I think `getProjectSize` would be easy to add here too. We also have to have `getOrganizationSize` but not sure if it's ready to go in here yet.

**Comment by tortillaj** [comment]
The two bigger things I think you'll want to change:

1) Adding tests (debatable if you actually want this 😄 )
2) You will want to move the guts of your route methods into separate functions that you can call them without restify - in particular so we can easily make CLI commands. For example, `getBuildById` would be a simple function that accepts an id, and your route method wraps it. This is what allows the CLI commands in the coordinator. (debatable about if you want to do this now, but i think we'll want it soon and it may be easier to start this way)

File: lib/Db.js (hunk 3 above)
Comment (lliss) [inline lib/Db.js]: See plugins for organization size.

File: lib/Api.js (hunk 2 above)
Comment (lliss) [inline lib/Api.js]: Sure. Updating...

Comment (lliss) [inline lib/Api.js]: Cleaned this up a good deal.

**Comment by lliss** [comment]
1. Added a bunch more test coverage. 
2. Re: CLI. The DB is already totally separate so we should be able to hit the same methods to connect via a CLI command when we want to add that.

**Pull request merged**